DEVELOPMENT.md

```diff
@@ -0,0 +1,136 @@
+
+## Checkout your fork
+
+The Go tools require that you clone the repository to the `src/github.com/GoogleContainerTools/skaffold` directory
+in your [`GOPATH`](https://github.com/golang/go/wiki/SettingGOPATH).
+
+To check out this repository:
+
+1. Create your own [fork of this
+  repo](https://help.github.com/articles/fork-a-repo/)
+
+1. Clone it to your machine:
+
+   ```shell
+   mkdir -p ${GOPATH}/src/github.com/GoogleContainerTools
+   cd ${GOPATH}/src/github.com/GoogleContainerTools
+   git clone git@github.com:${YOUR_GITHUB_USERNAME}/skaffold.git
+   cd skaffold
+   git remote add upstream git@github.com:GoogleContainerTools/skaffold.git
+   git remote set-url --push upstream no_push
+   ```
+
+   _Adding the `upstream` remote sets you up nicely for regularly [syncing your
+   fork](https://help.github.com/articles/syncing-a-fork/)._
+
+## Building skaffold
+
+To build with your local changes you have two options:
+
+1. Build the skaffold binary:
+
+   ```shell
+   make
+   ./out/skaffold version
+   ```
+
+   You can then run this binary directly, or copy/symlink it into your path.
+
+1. Build and install the skaffold binary:
+
+   ```shell
+   make install
+   skaffold version
+   ```
+
+   This will install skaffold via `go install` (note that if you have [manually downloaded
+   and installed skaffold to `/usr/bin/local`](README.adoc#installation), this is will probably
+   take precedence in your path over your `$GOPATH/bin`).
+
+   _If you are unsure if you are running a released or locally built version of skaffold, you
+   can run `skaffold version` - output which includes `dirty` indicates you have built the
+   binary locally._
+
+## Verifying local changes
+
+If you are iterating on skaffold and want to see your changes in action, you can:
+
+1. [Build skaffold](#building-and-running-skaffold)
+2. [Use the quickstart example](README.adoc#iterative-development)
+
+## Testing skaffold
+
+skaffold has both [unit tests](#unit-tests) and [integration tests](#integration-tests).
+
+### Unit Tests
+
+The unit tests live with the code they test and can be run with:
+
+```shell
+make test
+```
+
+_These tests will not run correctly unless you have [checked out your fork into your `$GOPATH`](#checkout-your-fork)._
+
+### Integration tests
+
+The integration tests live in [`integration`](./integration) and run the [`examples`](./examples)
+as tests. They can be run with:
+
+```shell
+make integration-test
+```
+
+_These tests require push access to a project in GCP, and so can only be run
+by maintainers who have access. These tests will be kicked off by [reviewers](#reviews)
+for submitted PRs._
+
+## Building skaffold docs
+
+Before [creating a PR](#creating-a-pr) with doc changes, we recommend that you locally verify the
+generated docs with:
+
+```shell
+make docs
+```
+
+And then open the generated docs/generated folder for `index.html` and `index.pdf`.
```

Comment:
One part of this process I'm still not clear on: now that I've made doc changes, do I need to do anything special to publish those?

Reply:
no, docs automatically get published after changes are merged to master in the latest builds bucket on https://storage.googleapis.com/skaffold/builds/latest/docs/index.html. When we do a release it gets released into the given versions bucket: https://storage.googleapis.com/skaffold/releases/latest/docs/index.html

Currently we don't host these on anywhere else than GCS.